metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -18,254 +28,1381 @@
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=dataset_urn,
+                aspectName="subTypes",
+                aspect=SubTypesClass(typeNames=["ML Models"]),
+            ),
+        )
+        self.report.report_workunit(subtypes_aspect)
+        yield subtypes_aspect
+        yield from self._get_domain_wu(
+            dataset_name=dataset_name,
+            entity_urn=dataset_urn,
+            entity_type="dataset",
+            sql_config=sql_config,
+        )
 
-    def __init__(self, timezone=False, precision=None):
-        """Construct a new :class:`_types.TIMESTAMP_WITH_PRECISION`.
+    def get_model_properties(
+        self, inspector: Inspector, schema: str, model: str
+    ) -> Tuple[Optional[str], Dict[str, str], Optional[str]]:
+        """
+        Returns ml models related metadata information to show in properties tab
+            eg. ml model attribute and ml model specification information.
 
-        :param timezone: boolean.  Indicates that the TIMESTAMP type should
-         enable timezone support, if available on the target database.
-         On a per-dialect basis is similar to "TIMESTAMP WITH TIMEZONE".
-         If the target database does not support timezones, this flag is
-         ignored.
-        :param precision: integer.  Indicates the PRECISION field when provided
+        Args:
+            inspector (Inspector): inspector obj from reflection engine
+            schema (str): schema name
+            model (str): ml model name
+        Returns:
+            Tuple[Optional[str], Dict[str, str], Optional[str]]
+        """
+        description: Optional[str] = None
+        properties: Dict[str, str] = {}
+        # The location cannot be fetched generically, but subclasses may override
+        # this method and provide a location.
+        location: Optional[str] = None
+        try:
+            # SQLAlchemy stubs are incomplete and missing this method.
+            # PR: https://github.com/dropbox/sqlalchemy-stubs/pull/223.
+            table_info: dict = inspector.get_model_comment(model, schema)  # type: ignore
+        except NotImplementedError:
+            return description, properties, location
+        except ProgrammingError as error:
+            logger.debug(
+                f"Encountered ProgrammingError. Retrying with quoted schema name for schema {schema} and {model} %s",
+                error,
+            )
+            table_info: dict = inspector.get_model_comment(model, f'"{schema}"')  # type: ignore
+        description = table_info.get("text")
+        if type(description) is tuple:
+            # Handling for value type tuple which is coming for dialect 'db2+ibm_db'
+            description = table_info["text"][0]
+        # The "properties" field is a non-standard addition to SQLAlchemy's interface.
+        properties = table_info.get("properties", {})
+        return description, properties, location
 
+    def loop_Oauth(
```

Comment:
```suggestion
    def loop_oauth(
```

Reply:
across the board, please use `oauth` instead of `Oauth`

Reply:
changed